generator/README.md

```diff
@@ -0,0 +1,1186 @@
+}
+
+- (nullable NSString *)hardware {
+    return [self.store sdl_objectForName:SDLRPCParameterNameHardware ofClass:NSString.class error:nil];
+}
+
+- (void)setFirmwareRev:(nullable NSString *)firmwareRev {
+    [self.store sdl_setObject:firmwareRev forName:SDLRPCParameterNameFirmwareRevision];
+}
+
+- (nullable NSString *)firmwareRev {
+    return [self.store sdl_objectForName:SDLRPCParameterNameFirmwareRevision ofClass:NSString.class error:nil];
+}
+
+- (void)setOs:(nullable NSString *)os {
+    [self.store sdl_setObject:os forName:SDLRPCParameterNameOS];
+}
+
+- (nullable NSString *)os {
+    return [self.store sdl_objectForName:SDLRPCParameterNameOS ofClass:NSString.class error:nil];
+}
+
+- (void)setOsVersion:(nullable NSString *)osVersion {
+    [self.store sdl_setObject:osVersion forName:SDLRPCParameterNameOSVersion];
+}
+
+- (nullable NSString *)osVersion {
+    return [self.store sdl_objectForName:SDLRPCParameterNameOSVersion ofClass:NSString.class error:nil];
+}
+
+- (void)setCarrier:(nullable NSString *)carrier {
+    [self.store sdl_setObject:carrier forName:SDLRPCParameterNameCarrier];
+}
+
+- (nullable NSString *)carrier {
+    return [self.store sdl_objectForName:SDLRPCParameterNameCarrier ofClass:NSString.class error:nil];
+}
+
+- (void)setMaxNumberRFCOMMPorts:(nullable NSNumber<SDLUInt> *)maxNumberRFCOMMPorts {
+    [self.store sdl_setObject:maxNumberRFCOMMPorts forName:SDLRPCParameterNameMaxNumberRFCOMMPorts];
+}
+
+- (nullable NSNumber<SDLUInt> *)maxNumberRFCOMMPorts {
+    return [self.store sdl_objectForName:SDLRPCParameterNameMaxNumberRFCOMMPorts ofClass:NSNumber.class error:nil];
+}
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+## Functions
+
+Functions in iOS implemented as 3 different classes grouped by its kind, all the 3 extend the common parent class **SDLRPCMessage**
+```objc
+// This is the parent class for all the 3 function kinds
+SDLRPCMessage : SDLRPCStruct <NSCopying> : NSObject <NSCopying>
+
+SDLRPCNotification : SDLRPCMessage  /// An RPC sent from the head unit to the app about some data change, such as a button was pressed
+SDLRPCRequest : SDLRPCMessage   /// Superclass of RPC requests
+SDLRPCResponse : SDLRPCMessage  /// Superclass of RPC responses
+```
+*Note: for some reason SDLRPCMessage conforms to NSCopying protocol twice.*
+
+First of all there is the **FunctionID** class generated though it is not declared in the XML. This class maps all function IDs that are integers to function names as strings.
+
+1. Uses of the `"name"` attribute should be normalized by the removal of the ID suffix, e.g. `RegisterAppInterfaceID -> RegisterAppInterface`. 
+  
+0. The constant name should be camel case formatted.
+  
+0. The constant has 2 fields the first is the `int` value of the `"value"` attribute and the second is the `String` value of normalized `"name"` attribute.
+
+Internally it uses another file that lists all the function names **SDLRPCFunctionNames**
+
+```objc
+//  SDLFunctionID.h
```

Comment:
The exact FunctionID has completely another structure that other enums, are you sure that it is good idea to put into documentation generated SDLFunctionID.h instead of existing one?

Reply:
I agree.

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/45f8005b9c7bd32e72f8c0891e85feb55d688d93